File: output/gemma4-e2b/avatar/Python/Java/atcoder_ABC042_A.java
import java.util.Scanner;
import java.util.Arrays;

public class atcoder_ABC042_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        // Read the line of input
        if (!scanner.hasNextLine()) {
            return;
        }
        String line = scanner.nextLine();
        
        // Parse the input into an array of integers (equivalent to tuple(map(int, input().split())))
        String[] parts = line.trim().split("\\s+");
        
        if (parts.length == 0) {
            System.out.println("NO");
            return;
        }
        
        int[] abc = new int[parts.length];
        try {
            for (int i = 0; i < parts.length; i++) {
                abc[i] = Integer.parseInt(parts[i]);
            }
        } catch (NumberFormatException e) {
            // Handle case where input is not purely integers, though assumed valid based on Python context
            System.out.println("NO");
            return;
        }

        // Check conditions
        int countOfFive = 0;
        long sum = 0;
        
        for (int x : abc) {
            sum += x;
            if (x == 5) {
                countOfFive++;
            }
        }

        if (countOfFive == 2) {
            if (sum == 17) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        } else {
            System.out.println("NO");
        }
        
        scanner.close();
    }
}

File: output/gemma4-e2b/avatar/Java/Python/atcoder_ABC042_A.py
import sys

try:
    # Read the line from standard input
    line = sys.stdin.readline().strip()
    if not line:
        # Handle empty input
        sys.exit()
        
    # Split the line into parts and convert to integers
    list_nums = line.split()
    if len(list_nums) < 3:
        # Handle case where input is insufficient
        sys.exit()
        
    x = int(list_nums[0])
    a = int(list_nums[1])
    y = int(list_nums[2])
    
    h = "NO"
    
    # Check the first condition: x == 5 | x == 7 | a == 5 | a == 7 | y == 5 | y == 7
    condition1 = (x == 5 or x == 7 or a == 5 or a == 7 or y == 5 or y == 7)
    
    if condition1:
        # Check the second condition: x + y + a == 17
        if x + y + a == 17:
            h = "YES"
            
    print(h)

except Exception:
    # In case of any error during reading or conversion
    pass